dev-local/msls/main.go

```diff
@@ -112,37 +118,71 @@ func handleLogin(rw http.ResponseWriter, r *http.Request) {
 	return &userinfo{
-		Sub:  u,
-		Hicn: p,
+		Sub:  usr,
+		Name: name,
+		Given_name: given_name,
+		Family_name: family_name,
+		Email: email,
+		Hicn: hicn,
+		Mbi: mbi,
+		Identity_type: identity_type,
 	}
 }
 
-func decode(c string) (string, string) {
-	u, _ := base64.RawURLEncoding.DecodeString(strings.Split(c, ".")[0])
-	p, _ := base64.RawURLEncoding.DecodeString(strings.Split(c, ".")[1])
-	return string(u), string(p)
+func decode(c string) (string, string, string, string, string, string, string, string) {
+	d_usr, _ := base64.RawURLEncoding.DecodeString(strings.Split(c, ".")[0])
+	d_name, _ := base64.RawURLEncoding.DecodeString(strings.Split(c, ".")[1])
+	d_given_name, _ := base64.RawURLEncoding.DecodeString(strings.Split(c, ".")[2])
+	d_family_name, _ := base64.RawURLEncoding.DecodeString(strings.Split(c, ".")[3])
+	d_email, _ := base64.RawURLEncoding.DecodeString(strings.Split(c, ".")[4])
+	d_hicn, _ := base64.RawURLEncoding.DecodeString(strings.Split(c, ".")[5])
+	d_mbi, _ := base64.RawURLEncoding.DecodeString(strings.Split(c, ".")[6])
+	d_identity_type, _ := base64.RawURLEncoding.DecodeString(strings.Split(c, ".")[7])
+	return string(d_usr), string(d_name), string(d_given_name), string(d_family_name),
+	       string(d_email), string(d_hicn), string(d_mbi), string(d_identity_type)
 }
 
-func encode(usr, pwd string) code {
-	u := base64.RawURLEncoding.EncodeToString([]byte(usr))
-	p := base64.RawURLEncoding.EncodeToString([]byte(pwd))
-	return code(fmt.Sprintf("%s.%s", u, p))
+func encode(usr,name, given_name, family_name, email, usr_identity, identity_type string) code {
+	e_usr := base64.RawURLEncoding.EncodeToString([]byte(usr))
+	e_name := base64.RawURLEncoding.EncodeToString([]byte(name))
+	e_given_name := base64.RawURLEncoding.EncodeToString([]byte(given_name))
+	e_family_name := base64.RawURLEncoding.EncodeToString([]byte(family_name))
+	e_email := base64.RawURLEncoding.EncodeToString([]byte(email))
+	e_identity_type := base64.RawURLEncoding.EncodeToString([]byte(identity_type))
+	e_hicn := ""
+	if identity_type == "H" {
+		e_hicn = base64.RawURLEncoding.EncodeToString([]byte(usr_identity))
+	}
+	e_mbi := ""
+	if identity_type == "M" {
+		e_mbi = base64.RawURLEncoding.EncodeToString([]byte(usr_identity))
+	}
+	return code(fmt.Sprintf("%s.%s.%s.%s.%s.%s.%s.%s", e_usr, e_name, e_given_name,
+				e_family_name, e_email, e_hicn, e_mbi, e_identity_type))
 }
 
 type userinfo struct {
-	Sub        string `json:"sub"`
-	Hicn       string `json:"hicn"`
-	GivenName  string `json:"given_name"`
-	FamilyName string `json:"family_name"`
-	Email      string `json:"email"`
+	Sub         	string `json:"sub"`
+	Name        	string `json:"name"`
+	Given_name  	string `json:"given_name"`
+	Family_name 	string `json:"family_name"`
+	Email       	string `json:"email"`
+	Hicn   			string `json:"hicn"`
+	Mbi   			string `json:"mbi"`
+	Identity_type	string `json:"identity_type"`
```

Comment:
See previous comment. `Identify_type` is not a value that the SLS `userinfo` end point returns.

Reply:
removed